src/SB/Core/x/xPtankPool.cpp

```diff
@@ -242,8 +343,101 @@ void xPTankPoolRender()
 
             g->used = 0;
             sort_buckets(*g);
-            
         }
         g++;
     }
 }
+
+void ptank_pool::grab_block(ptank_group_type type)
+{
+    this->ptank = NULL;
+    group_data& group = groups[type];
+    if (group.used < group.size || create_ptanks(group, 1) >= 1)
+    {
+        ptank_context** end = group.buckets + group.buckets_used;
+
+        {
+            ptank_context** it = group.buckets + 2;
+            while (it < end)
+            {
+                if (*it != NULL)
+                {
+                    if ((*it)->ptank->geometry->matList.materials[0]->texture == this->rs.texture &&
+                        (*it)->dst_blend == this->rs.dst_blend &&
+                        (*it)->src_blend == this->rs.src_blend &&
+                        !(((*it)->flags ^ this->rs.flags) & 0x1))
+                    {
+                        this->ptank = (*it)->ptank;
+                        (*it)->flags |= rpPTANKDFLAGCNSNORMAL;
+                        *it = (*it)->next;
+                        group.used++;
+                        return;
+                    }
+                }
+                it++;
+            }
+        }
+
+        {
+            ptank_context** it = group.buckets;
+            while (it < end)
+            {
+                if (*it != NULL)
+                {
+                    if ((this->rs.flags & rpPTANKDFLAGPOSITION) &&
+                        !((*it)->flags & rpPTANKDFLAGPOSITION))
+                    {
+                        destroy_ptank((*it)->ptank);
+                        (*it)->ptank = create_ptank(group.create_flags);
+                    }
+                    this->ptank = (*it)->ptank;
+                    RpMaterialSetTexture(*this->ptank->geometry->matList.materials,
+                                         this->rs.texture);
+                    RPATOMICPTANKPLUGINDATA(this->ptank)->publicData.srcBlend = this->rs.src_blend;
+                    RPATOMICPTANKPLUGINDATA(this->ptank)->publicData.dstBlend = this->rs.dst_blend;
+                    RPATOMICPTANKPLUGINDATA(this->ptank)->instFlags |= rpPTANKDFLAGARRAY;
+
+                    (*it)->src_blend = this->rs.src_blend;
+                    (*it)->dst_blend = this->rs.dst_blend;
+                    (*it)->flags = this->rs.flags | rpPTANKDFLAGCNSNORMAL;
+                    *it = (*it)->next;
+                    group.used++;
+                    return;
+                }
+
+                it++;
+            }
+        }
+    }
+}
+
+void ptank_pool::flush()
+{
+    if (this->valid() != 0)
```

Comment:
Does this also match? If so, please use the suggestion instead, as it's a bit cleaner.
```suggestion
    if (this->valid())
```

Reply:
As you probably saw, I also removed all comparisons to zero in the file.

Reply:
As you probably saw, I removed a few other direct comparisons to zero in the file